lib/ruby2d/window.rb

```diff
@@ -26,14 +27,19 @@ def initialize(args = {})
       @vsync       = args[:vsync]  || true
       @mouse_x, @mouse_y = 0, 0
       @objects     = []
-      @key, @key_down, @key_held, @key_up = [], [], [], []
-      @mouse, @mouse_down, @mouse_up, @mouse_scroll, @mouse_move = [], [], [], [], []
-      @controller, @controller_axis = [], []
-      @controller_button_down, @controller_button_up = [], []
+      @key, @key_down, @key_held, @key_up = {}, {}, {}, {}
+      @mouse, @mouse_down, @mouse_up, @mouse_scroll, @mouse_move = {}, {}, {}, {}, {}
+      @controller, @controller_axis = {}, {}
+      @controller_button_down, @controller_button_up = {}, {}
+      @event_key = "a"
```

Comment:
Since `@event_key` is used as an ID, what do think about making this a number, like starting at `0` and calling `@event_key.next` in the `new_event_key` method. Just might be clearer.